### PR TITLE
Casmpet 6008: Kyverno deployment needs anti-affinity set to required, not preferred & add pod priority

### PR DESCRIPTION
**Pull request by pradeepkumargl**
This pull request is for fixing CASMPET-6008/CASMTRIAGE-4221 issue.

### CODE REVIEW

**Comment by yarlagadda-srinivasa-rao** [comment]
We have successfully validated and verified the changed parameters in the Cray-Kyverno chart on mug system.
As this issue is a corner case and it is not reproducible,  in addition there are no upgrade tests planned today during India time, we have not included the test logs.
Since similar changes are already implemented for other charts in CSM, with that confidence and based on our successful sanity tests on mug, we have raised the PR for approval.

Test log from Mug System:
ncn-m001:/tmp/test1 # helm history -n kyverno cray-kyverno
REVISION        UPDATED                         STATUS          CHART                   APP VERSION     DESCRIPTION
1               Thu Sep 15 17:54:57 2022        superseded      cray-kyverno-1.2.0      v1.6.2          Install complete
2               Thu Sep 15 18:13:48 2022        superseded      cray-kyverno-1.2.0      v1.6.2          Upgrade complete
3               Fri Sep 23 04:58:59 2022        superseded      cray-kyverno-1.2.0      v1.6.2          Upgrade complete
4               Fri Sep 23 05:12:32 2022        superseded      cray-kyverno-1.2.0      v1.6.2          Upgrade complete
5               Fri Sep 23 05:22:27 2022        superseded      cray-kyverno-1.2.0      v1.6.2          Upgrade complete
6               Fri Sep 23 07:25:22 2022        deployed        cray-kyverno-1.2.0      v1.6.2          Rollback to 2
cat cray-kyverno/charts/kyverno/values.yaml
....
antiAffinity:
  # using this option will schedule pods on different nodes if possible
  enable: true

podAntiAffinity:
  requiredDuringSchedulingIgnoredDuringExecution:
    - labelSelector:
        matchExpressions:
        - key: app.kubernetes.io/name
          operator: In
          values:
          - kyverno
      topologyKey: kubernetes.io/hostname

priorityClassName: csm-high-priority-service

podAffinity: {}

nodeAffinity: {}
...

**Review by bklei — APPROVED** [review]

**Review by bo-quan — APPROVED** [review]

**Comment by djlapenta** [comment]
@denniswalker  -  if we want to hold this for an upgrade test, we can..  but we need to make sure we put the special instructions in the installer assignments for Monday to indicate that the fixed kyverno bits need to be replaced --before we run prerequisites in the upgrade--  because kyverno gets installed in pre-reqs.

**Pull request merged**
